### DISCUSSION TITLE
Listing Possible Sources for information

### DISCUSSION THREAD

**Discussion by alexvonme** [jargonLint/jargonLint#31] [Ideas]
As we discussed earlier, I'm collating everything regarding possible sources of information here. I've added some of our comments regarding each.

| Source                    | Link                                                                                     | License   | Comments                                                                                                           |
|---------------------------|------------------------------------------------------------------------------------------|-----------|--------------------------------------------------------------------------------------------------------------------|
| libraries.io              | https://github.com/librariesio/libraries.io                                              | CC-BY-4.0 | It may provide valuable resources of words and cases. Parsing the API results into yaml list could be interesting. |
| libraries.io (2)          | https://github.com/hackebrot/go-librariesio                                              | MIT       | This lib might be interesting even if code is very old                                                             |
| cspell                    | https://github.com/streetsidesoftware/cspell-dicts/tree/main/dictionaries/software-terms | GPL 3     | Related discussion: streetsidesoftware/cspell-dicts#3310 (comment)                                                 |
| frontend-encyclopedia     | https://github.com/paradite/frontend-encyclopedia                                        | MIT       |                                                                                                                    |
| glossario                 | https://github.com/carpentries/glosario                                                  | CC-BY-4.0 |                                                                                                                    |
| awesome/wordlists         | https://github.com/gmelodie/awesome-wordlists                                            | MIT-like  | The localization is detailed in https://github.com/en-wl/wordlist/tree/master/varcon                               |
| cheatsnake/backend-cheats | https://github.com/cheatsnake/backend-cheats                                             | MIT       | This file is wonderful, it could be a source of content for description but also URLs                              |
| shields.io                | https://github.com/badges/shields                                                        | CC0-1.0   |                                                                                                                    |
| alexandresanlim/Badges4   | https://github.com/alexandresanlim/Badges4-README.md-Profile                             | N/A?      | I opened the following issue to ask owner to create a license: alexandresanlim/Badges4-README.md-Profile#674       |
| SCOWL                     | https://github.com/en-wl/wordlist                                                        | MIT-like  | Generating dictionaries could be very useful.                                                                      |
| Qwarkus | https://github.com/quarkusio/quarkus/blob/main/docs/.vale/styles/Quarkus/CaseSensitiveTerms.yml | Apache 2.0 | a lot of redhat terms |
| [Canonical Praecepta](https://github.com/canonical/praecepta) | https://github.com/canonical/praecepta/blob/main/styles/config/vocabularies/Canonical/Accept.txt | CC-BY-4.0 |  Ubuntu words |
| awesome-auth | https://github.com/casbin/awesome-auth |  CC0 1.0 Universal | SSO tools and terms |
| cspell companies dictionary | [cspell-dicts/company](https://github.com/streetsidesoftware/cspell-dicts/tree/main/dictionaries%2Fcompanies%2Fdict) | MIT | many companies |

Let's discuss each with threads under this discussion topic.

You can check for licenses from this [Wikipedia article.](https://en.wikipedia.org/wiki/Comparison_of_free_and_open-source_software_licenses)

**Comment by ccoVeille** [comment]
What about Quarkus vale rules as a source

https://github.com/quarkusio/quarkus/blob/main/docs/.vale/styles/Quarkus/CaseSensitiveTerms.yml

License Apache 2.0

**Reply by ccoVeille** [reply]
Same license but limited list

https://github.com/saberder/docker.github.io/blob/89bdf57ecdaa2da9db4b99dd86a494ede47922c2/.github/vale/Vocab/Industry/accept.txt

**Reply by alexvonme** [reply]
It's a Red Hat source, they have a lot of Red Hat naming so that's useful.

I don't think the docker one is very usefull, its limited and we already added a bunch of those

**Reply by ccoVeille** [reply]
OK thanks.

I will open a PR when I have time.

We will have to work on tooling any to validate the rules, like the good/bad.txt on suse repository

**Reply by ccoVeille** [reply]
I edited (quick and dirty) to add qwarkus to the markdown table

**Comment by ccoVeille** [comment]
Can you edit your post to add the License that we cannot accept for the projecs?

It would be easier for me to look at it, and learn than looking at project to then ask you if we can deal with them

> give a man a fish and you feed him for a day; teach a man to fish and you feed him for a lifetime

**Reply by alexvonme** [reply]
> You can check for licenses from this [Wikipedia article.](https://en.wikipedia.org/wiki/Comparison_of_free_and_open-source_software_licenses)

Done. We currently prioritize those with licensing where modification is permissive and/or public domain.

**Reply by ccoVeille** [reply]
Very useful link, thanks for giving me idea what to look for

**Comment by ccoVeille** [comment]
Just to make sure this repository which is GNU General Public license 3.0

https://github.com/streetsidesoftware/cspell-dicts/tree/main/dictionaries%2Fcompanies%2Fdict

Cannot be imported?

**Reply by alexvonme** [reply]
cspell is great, I don't think we should disregard it but we definitely need to be extra careful with copyleft licenses.

**Reply by ccoVeille** [reply]
Theorical question: do you think the maintainer could change its license, or at least the license of the cspell-dicts, not necessarily the license of the cspell (code) ? or is it impossible once you are in GPL3, you are stuck with it ?

I'm asking because I chatted with him a few times

**Reply by ldemailly** [reply]
if they are the author they could dual license but doubt they will, for something I got dictionaries from linux distros (eg wamerican-large) but you could probably have GPL3 as sub module or some such if needed?

**Reply by alexvonme** [reply]
Likewise, I doubt they'll dual license even if they can. GPL3 doesn't affect anything else when it's a YAML file so we shouldn't have problems. We just need to clarify that those rules and their iterations are GPL3 licensed.

**Reply by ccoVeille** [reply]
In fact, there is no need

The company dictionary is under MIT license

https://github.com/streetsidesoftware/cspell-dicts/blob/main/dictionaries%2Fcompanies%2FLICENSE

Only the code is the project is under GPLv3

**Reply by ccoVeille** [reply]
You are awesome @jason3s

**Comment by ccoVeille** [comment]
I added [Praecepta](https://github.com/canonical/praecepta) to the doc, while the repository is still small, it's a good license and official ubuntu vale

**Comment by ccoVeille** [comment]
I found https://github.com/osmlab/name-suggestion-index (BSD-3)

It provides for example brands in their canonical names.

I'm unsure there is interesting content in it, but I'm adding the link for future investigation and reference.

Only a small part maybe used for jargonLint

**Reply by ccoVeille** [reply]
Stepping out jargonLint for a bit

@jason3S is there anything interesting for cspell-dicts brand dictionary in this project?

https://github.com/osmlab/name-suggestion-index

**Reply by Jason3S** [reply]
@ccoVeille,

Nice dataset! It seems to be a great collection of Points of Interest.

I'm not sure if it is immediately useful for the spell checker. It would be a large dictionary and would contain many misspellings (both accidental and deliberate). Even so, I will keep it in mind, it might be a good reference.

**Comment by ccoVeille** [comment]
https://github.com/sindresorhus/awesome-lint/blob/main/lib/spell-check-rules.js MIT license
